collector/dns.go

```diff
@@ -0,0 +1,492 @@
+type DNSCollector struct {
+	ZoneTransferRequestsReceived  *prometheus.Desc
+	ZoneTransferRequestsSent      *prometheus.Desc
+	ZoneTransferResponsesReceived *prometheus.Desc
+	ZoneTransferSuccessReceived   *prometheus.Desc
+	ZoneTransferSuccessSent       *prometheus.Desc
+	ZoneTransferFailures          *prometheus.Desc
+	MemoryUsedBytes               *prometheus.Desc
+	DynamicUpdatesQueued          *prometheus.Desc
+	DynamicUpdatesReceived        *prometheus.Desc
+	DynamicUpdatesFailures        *prometheus.Desc
+	NotifyReceived                *prometheus.Desc
+	NotifySent                    *prometheus.Desc
+	SecureUpdateFailures          *prometheus.Desc
+	SecureUpdateReceived          *prometheus.Desc
+	Queries                       *prometheus.Desc
+	Responses                     *prometheus.Desc
+	RecursiveQueries              *prometheus.Desc
+	RecursiveQueryFailures        *prometheus.Desc
+	RecursiveQuerySendTimeouts    *prometheus.Desc
+	WinsQueries                   *prometheus.Desc
+	WinsResponses                 *prometheus.Desc
+	UnmatchedResponsesReceived    *prometheus.Desc
+}
+
+// NewDNSCollector ...
+func NewDNSCollector() (Collector, error) {
+	const subsystem = "dns"
+	return &DNSCollector{
+		ZoneTransferRequestsReceived: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "zone_transfer_requests_received_total"),
+			"Number of zone transfer requests (AXFR/IXFR) received by the master DNS server",
+			[]string{"type"},
+			nil,
+		),
+		ZoneTransferRequestsSent: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "zone_transfer_requests_sent_total"),
+			"Number of zone transfer requests (AXFR/IXFR) sent by the secondary DNS server",
+			[]string{"type"},
+			nil,
+		),
+		ZoneTransferResponsesReceived: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "zone_transfer_response_received_total"),
+			"Number of zone transfer responses (AXFR/IXFR) received by the secondary DNS server",
+			[]string{"type"},
+			nil,
+		),
+		ZoneTransferSuccessReceived: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "zone_transfer_success_received_total"),
+			"Number of successful zone transfers (AXFR/IXFR) received by the secondary DNS server",
+			[]string{"type", "protocol"},
+			nil,
+		),
+		ZoneTransferSuccessSent: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "zone_transfer_success_sent_total"),
+			"Number of successful zone transfers (AXFR/IXFR) of the master DNS server",
+			[]string{"type"},
+			nil,
+		),
+		ZoneTransferFailures: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "zone_transfer_failures_total"),
+			"Number of failed zone transfers of the master DNS server",
+			nil,
+			nil,
+		),
+		MemoryUsedBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "memory_used_bytes_total"),
+			"Total memory used by DNS server, per type",
+			[]string{"type"},
```

Comment:
Here's another `type` label which should be changed. Possible values are caching/database_node/nbstat/record_flow/tcp_message/udp_message. 
`pool` is a common label name, and while probably not very correct it might still be better. Other suggestions? Area?